models/DataObject/Concrete.php

```diff
@@ -736,11 +724,9 @@ public function __wakeup(): void
         parent::__wakeup();
 
         // renew object reference to other object aware fields
-        foreach ($this->__objectAwareFields as $objectAwareField => $exists) {
-            if (isset($this->$objectAwareField)) {
-                if ($this->$objectAwareField instanceof ObjectAwareFieldInterface) {
-                    $this->$objectAwareField->setObject($this);
-                }
+        foreach (get_object_vars($this) as $propertyValue) {
```

Comment:
We wanted to avoid iterating through all class variables due to performance reasons. 
But I guess, there is better option to do so, right? 

The actual problem was caused by removing the extra if for localized fields [here](https://github.com/pimcore/pimcore/pull/13874/files#diff-8ff96247c8a26d5a3ecbf9b79e38c31b8fc8a638bd145a46b9cdb7ee27d4869bL761)

@brusch wdyt?

Reply:
I was thinking about an 

```
// original
foreach ($this->__objectAwareFields as $objectAwareField => $exists) {
// change to
foreach ([...$this->__objectAwareFields, 'localizedfields' => true] as $objectAwareField => $exists) {
```

at first, but then figured that this same bug can occur in the future when one of the other fields types is marked with the `ObjectAwareFieldInterface` (where it wasn't at the moment that the object is serialized).

But I do get your performance reasons, especially because the `__wakeup()` is also triggered when data is loaded from the Redis (or other) cache.

As alternative, we could also introduce a method
```
class Concrete {
    /**
     * Resolve issues that might occur due to old serialized data
     */
    public function onVersionDataLoaded(): void
    {
        // renew object reference to other object aware fields
        // similar as __wakeup() but now test ALL fields
        foreach (get_object_vars($this) as $propertyValue) {
            if ($propertyValue instanceof ObjectAwareFieldInterface) {
                $propertyValue->setObject($this);
            }
        }
    }
}
```
and trigger this method only when a serialized version is loaded (and not when an object is loaded from cache)